src/pages/home/report/ReportActionPropTypes.js

```diff
@@ -9,8 +9,12 @@ export default {
     // Person who created the action
     person: PropTypes.arrayOf(ReportActionFragmentPropTypes).isRequired,
 
-    // ID of the report action
-    sequenceNumber: PropTypes.number.isRequired,
+    // ID of the report action. Can be either a number or a string since
+    // temporary report action IDs are unique strings generated in the client.
+    sequenceNumber: PropTypes.oneOfType([
+        PropTypes.number,
+        PropTypes.string,
```

Comment:
Sorry if I missed a convo, but can we make these always be strings?

Reply:
I'll have to look into that. But my first thought is that there are a lot of cases where they are not strings so they would need to be cast to strings only to make the `PropTypes` checker happy. Maybe that would resolve whatever concern there is here? And just to clarify... what is the concern?

Reply:
We discussed this on e.cash, imo this component should accept a single type for sequenceNumber

Reply:
Yes, but I think the request here requires more consideration (I'll do my best to explain why). I agree we should solve it in a holistic way, but changing the `propTypes` here doesn't exactly achieve the desired outcome. For now, this accurately reflects what the data in Onyx can be and therefore accurately reflects the current state of things. Casting to a string would need to be done at a low level (in the Onyx data).

e.g. the `reportActions` here would need to have all `sequenceNumber` converted to strings in the Onyx data.

https://github.com/Expensify/Expensify.cash/blob/792277063ed8049952785112318ff670de5dbc31/src/pages/home/report/ReportActionsView.js#L24-L27

But if we coerce them to strings there could be unexpected results in other places (like `actions/Report.js`) where we expect `sequenceNumber` to be a number. We are also only using the `sequenceNumber` in the UI to create the React `key`.

I am less sure about casting all `sequenceNumber` to strings in storage and we can look into whether this is a safe change or not. But my preference would be to create a new issue and start sending string `sequenceNumber` at the API level if it makes sense to do that.

Reply:
Thanks for the explanation.

Tbh, I'm not a huge fan, especially since this was accepting only numbers prior to this PR, so this feels like we're either introducing technical debt that we need to undo, or just adding on to existing one. 

That said, I don't want to block so go ahead with the new issue (and fwiw, I'd love for it to be high on your priority list)

Reply:
Cool. Let's discuss more in the issue? https://github.com/Expensify/Expensify/issues/152504